### PR TITLE
Text edit

### PR DESCRIPTION
**Pull request by fsih**
Progress toward https://github.com/LLK/scratch-paint/issues/25
Adds text editing to the text tool!

See diff from initial text tool PR here:
https://github.com/fsih/scratch-paint/compare/textTool...fsih:textEdit?expand=1

When text is in edit mode, add a text area directly on top of it. This way we get all text editing capabilities of text areas.

Some notes
- The text edit area needs to stay on top of the text appearing on the canvas, no matter what orientation it's in. If the text area is able to be scrolled or wraps, it will be out of sync. The text can also have some crazy transformations applied to it through the select tool. This math is tricky, so this should be thoroughly tested.
- Zooming in and scrolling around the canvas also changes the matrix of the text area. I needed to add state to keep track of when that happened (state.scratchPaint.viewBounds)
- Added undo state to the text tool. When the user stops typing for 1 second, a new undo state is created. This is a pretty long timeout. I made it this long because sometimes kids aren't familiar with keyboard layouts. We might want to play with this number.

### CODE REVIEW

**Comment by fsih** [comment]
I made the text tool draw no outline by default, as recommended by Paul

**Comment by paulkaplan** [comment]
Just a few things I noticed playing with this. I'll probably create separate issues after this is merged, but wanted to have them down here for posterity. 

When clicking another text area while editing a text area, the currently editing text area should both shift out of editing mode (which it does) _and_ shift out of select mode (which it does not).
text-select-1

When in text editing mode, after exiting a text area (so that it is selected but not editing), you cannot re-enter editing on that text area without clicking outside to deselect it first
text-select-2

When in the normal select mode, there is a heavy expectation that double clicking on text would enter editing mode (from most GUIs with text areas, including scratch 2), but it doesn't. 
text-select-3

When in normal select mode, with a text area selected, clicking the text mode tool in scratch 2 transitions that text area into editing mode. Here it simply deselects everything. In the case where 
 multiple things are selected, scratch 2 deselects everything. 
text-select-4

**Comment by paulkaplan** [comment]
Fixed the first and second issue with a commit: af2c49e

File: src/helper/tools/text-tool.js [hunk 1]
```diff
@@ -0,0 +1,326 @@
...
+    }
+    handleMouseMove (event) {
+        const hitResults = paper.project.hitTestAll(event.point, this.getTextEditHitOptions());
+        if (hitResults.length) {
+            document.body.style.cursor = 'text';
+        } else {
+            document.body.style.cursor = 'auto';
+        }
+    }
+    handleMouseDown (event) {
```
Comment (paulkaplan) [inline src/helper/tools/text-tool.js:138]: There is some seriously complicated logic in this function. I'd like for us to come back to this and break it into more clear state transitions. It took me ~20 minutes to understand the entire FROM => TO transitions!

**Review by paulkaplan — APPROVED** [review]
This LG for now. There are obviously some follow ups we'll need to make (a few were listed in the comments) but this feels good enough to start testing with.

**Pull request merged**